plugin/trino-example-plugin/src/main/java/com/example/ExampleMetadata.java

```diff
@@ -0,0 +1,144 @@
+                    new ColumnMetadata("name", VARCHAR)))
+            .build();
+
+    @Inject
+    public ExampleMetadata()
+    {
+    }
+
+    @Override
+    public List<String> listSchemaNames(ConnectorSession connectorSession)
+    {
+        return List.of(SCHEMA_NAME);
+    }
+
+    @Override
+    public ConnectorTableHandle getTableHandle(ConnectorSession connectorSession, SchemaTableName schemaTableName)
+    {
+        if (!schemaTableName.getSchemaName().equals(SCHEMA_NAME)) {
+            return null;
+        }
+        return new ExampleTableHandle(schemaTableName);
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata(
+            ConnectorSession connectorSession,
+            ConnectorTableHandle connectorTableHandle)
+    {
+        ExampleTableHandle tableHandle = checkType(connectorTableHandle, ExampleTableHandle.class, "tableHandle");
+        SchemaTableName schemaTableName = tableHandle.getSchemaTableName();
+        return new ConnectorTableMetadata(
+                schemaTableName,
+                columns.get(schemaTableName.getTableName()));
+    }
+
+    @Override
+    public List<SchemaTableName> listTables(ConnectorSession session, Optional<String> schemaName)
+    {
+        return columns
+                .keySet()
+                .stream()
+                .map(table -> new SchemaTableName(SCHEMA_NAME, table))
+                .collect(toList());
+    }
+
+    @Override
+    public Map<String, ColumnHandle> getColumnHandles(
+            ConnectorSession connectorSession,
+            ConnectorTableHandle connectorTableHandle)
+    {
+        return getTableMetadata(connectorSession, connectorTableHandle).getColumns().stream()
+                .collect(toMap(ColumnMetadata::getName, column -> new ExampleColumnHandle(column.getName(), column.getType())));
+    }
+
+    @Override
+    public ColumnMetadata getColumnMetadata(
+            ConnectorSession connectorSession,
+            ConnectorTableHandle connectorTableHandle,
+            ColumnHandle columnHandle)
+    {
+        ExampleColumnHandle restColumnHandle = checkType(columnHandle, ExampleColumnHandle.class, "columnHandle");
+        return new ColumnMetadata(restColumnHandle.getName(), restColumnHandle.getType());
+    }
+
+    @Override
+    public ConnectorTableProperties getTableProperties(ConnectorSession session, ConnectorTableHandle table)
+    {
+        return new ConnectorTableProperties();
+    }
+
+    @Override
+    public Iterator<TableColumnsMetadata> streamTableColumns(ConnectorSession session, SchemaTablePrefix prefix)
+    {
+        return columns.entrySet().stream()
+                .map(entry -> TableColumnsMetadata.forTable(
+                        new SchemaTableName(prefix.getSchema().orElse(""), entry.getKey()),
+                        entry.getValue()))
+                .iterator();
+    }
+
+    private static <A, B extends A> B checkType(A value, Class<B> target, String name)
```

Comment:
We don't do this anymore. Just use a normal cast. The JVM now produces a good error message.

Also, this would only happen if there was a serious bug in the engine, which is not something that connectors need to be defensive against.